spyderlib/plugins/shortcuts.py

```diff
@@ -7,179 +7,505 @@
+                self.npressed = 0
+
+    def check_conflicts(self):
+        """Check shortcuts for conflicts"""
+        conflicts = []
+        for index, shortcut in enumerate(self.shortcuts):
+            sequence = str(shortcut.key)
+            if sequence == self.new_sequence and \
+                (shortcut.context == self.context or shortcut.context == '_' or
+                 self.context == '_'):
+                conflicts.append(shortcut)
+        return conflicts
+
+    def update_warning(self, reset=False):
+        """ """
+        conflicts = self.check_conflicts()
+        if reset:
+            conflicts = []
+
+        widget = self.helper_button
+
+        if conflicts:
+            tip_title = _('The new entered shorcut conflicts with:') + '\n'
+            tip_body = ''
+            for s in conflicts:
+                tip_body += ' - {0}: {1}\n'.format(s.context, s.name)
+            tip = '{0}{1}'.format(tip_title, tip_body)
+            widget.setIcon(get_std_icon('MessageBoxWarning'))
+            widget.setToolTip(tip)
+            QToolTip.showText(widget.mapToGlobal(QPoint(0, 5)), tip)
+        else:
+            widget.setToolTip('')
+            QToolTip.hideText()
+            widget.setIcon(get_std_icon('DialogApplyButton'))
+
+    def set_sequence(self, sequence):
+        """Set the new shortcut and update buttons."""
+        if self.sequence != sequence:
+            self.button_ok.setEnabled(True)
+            reset = False
+        else:
+            self.button_ok.setEnabled(False)
+            reset = True
+
+        self.text_new_sequence.setText(sequence)
+        self.new_sequence = sequence
+        self.update_warning(reset=reset)
+
+    def validate_sequence(self):
+        """Provide additional checks for accepting or rejecting shortcuts."""
+        for mod in MODIFIERS:
+            non_mod = set(self.key_non_modifiers)
+            non_mod.discard(mod)
+            if mod in self.key_non_modifiers:
+                self.key_non_modifiers.remove(mod)
+
+        self.key_modifiers = self.key_modifiers - non_mod
+
+        while u'' in self.key_text:
+            self.key_text.remove(u'')
+
+        self.key_text = [k.upper() for k in self.key_text]
+
+        # Fix Backtab, Tab issue
+        if os.name == 'nt':
+            if Qt.Key_Backtab in self.key_non_modifiers:
+                idx = self.key_non_modifiers.index(Qt.Key_Backtab)
+                self.key_non_modifiers[idx] = Qt.Key_Tab
+
+        if len(self.key_modifiers) == 0:
+            # Filter single key allowed
+            if self.key_non_modifiers[0] not in VALID_SINGLE_KEYS:
+                return
+            # Filter
+            elif len(self.key_non_modifiers) > 1:
+                return
+
+        # QKeySequence accepts a maximum of 4 different sequences
+        if len(self.keys) > 4:
+            # If the user enters more than 4, take the first 4 only
+            self.keys = set([0, 1, 2, 3])

```

Comment:
I don't agree with this behavior. I think we shouldn't allow a shortcut in this case.